fixes react controlled input error

dimitropoulos opened the pull request:
closes INS-703

dimitropoulos committed:
fix ts-expect-error

dimitropoulos committed:
removes useless await

see https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/await-thenable.md

We're not quite ready to have this rule on everywhere, but in this case I can see that it's correct to follow.

dimitropoulos committed:
add comment about magic number

and moves the assignment to the one place it's used

I opted not to make a ticket because we are recently running out of tickets and this seems to be a relatively small thing @reynolek please confirm this is the correct course of action.

dimitropoulos committed:
initialize ref to reflect reality (and include actual type)

dimitropoulos committed:
use onChange instead of onInput to access correct types

ultimately, since react's behavior diverged from the DOM behavior to match onInput, this should have no effect on any functionality since react assigns this event to `onChange` anyway.  see https://github.com/facebook/react/blob/master/fixtures/attribute-behavior/src/attributes.js#L2089

dimitropoulos committed:
clears remaining ts-expect-errors in file

dimitropoulos commented on packages/insomnia-app/app/ui/components/modals/workspace-environments-edit-modal.tsx:
```diff
@@ -348,24 +341,16 @@ class WorkspaceEnvironmentsEditModal extends PureComponent<Props, State> {
+  _handleClickColorChange(environment: Environment) {
     if (!environment.color) {
-      await this._handleChangeEnvironmentColor(environment, color);
+      // TODO: fix magic-number. Currently this is the `suprise` background color for the default theme, but we should be grabbing the actual value from the user's actual theme instead.
+      const color = environment.color || '#7d69cb';
+      this._handleChangeEnvironmentColor(environment, color);
```
```suggestion
      // TODO: fix magic-number. Currently this is the `surprise` background color for the default theme, but we should be grabbing the actual value from the user's actual theme instead.
      this._handleChangeEnvironmentColor('#7d69cb');
```

develohpanda commented on packages/insomnia-app/app/ui/components/modals/workspace-environments-edit-modal.tsx:
```diff
@@ -508,16 +498,17 @@ class WorkspaceEnvironmentsEditModal extends PureComponent<Props, State> {
                     <DropdownItem
                       value={activeEnvironment}
-                      onClick={this._handleUnsetColor}
+                      onClick={() => {
+                        this._handleChangeEnvironmentColor(activeEnvironment, null);
+                      }}
```
Could you please provide context into this change (class function -> inline arrow function)?

dimitropoulos replied:
In the commit message I mentioned "inlined one-liners". The reason for doing this was to reduce all the indirection so I could tell what was actually going on.  I put some of them back, but I left the one-line simple ones.  In this case, the main reason for doing this was that we are calling `this._getActiveEnvironment()` twice in a row just for the ceremony of preventing re-renders when (I would wager, but have not measured) the database call and subsequent array search that occurs to do so far outweighs whatever performance gain we got from the avoided re-render, which, by the way, would be solved by memoizing the callback anyway.
 
Still, in the interest of moving forward, I've reverted this in e801afea4f64ffce803e17ad8fb059d28bb3a295

dimitropoulos committed:
updates fallback call per review feedback

dimitropoulos committed:
fixes root cause of error, inlines one-liners

the error with the type of `_handleInputColorChange` was due to `onChange` being the wrong prop to use (where `onChange` is the one we want).
